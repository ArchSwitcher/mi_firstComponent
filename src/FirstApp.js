import React, { Fragment } from 'react';
import PropPytes from 'prop-types';

const FirstApp = ( {greeting, subtitle} ) => {
    return (
        <>
            <h1>{greeting}</h1>
            {/* <pre> {JSON.stringify(greeting, null, 3)} </pre> */}
            <p> {subtitle}</p>
        </>
    );
}

FirstApp.propTypes = {
    greeting: PropPytes.string.isRequired

}

FirstApp.defaultProps = {
subtitle: 'im a subtitle'
}

export default FirstApp;    