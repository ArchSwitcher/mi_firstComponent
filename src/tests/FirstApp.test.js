// import { render } from "@testing-library/react";
import React from 'react';
import { shallow } from 'enzyme/build';
import '@testing-library/jest-dom';
import FirstApp from "../FirstApp"


describe('make tests in component FirstApp', () => {
    // test('should be retrun message "Hi i\'m Arch Switcher" ', () => {
    // const greeting = 'Hi i\'m Arch Switcher';
    // const { getByText } = render( < FirstApp greeting = { greeting }
    //     /> );
    //     expect(getByText(greeting)).toBeInTheDocument;
    // });
    // });

    test('should be show <FirstApp /> success', () => {

        // shallow es similar a render pero da otras opciones interesantes simular clicks y otras opciones;
        // const greeting = 'Hi i\'m Arch Switcher';

        const wrapper = shallow( < FirstApp greeting = 'Hi im Arch Switcher' / > );
        expect(wrapper).toMatchSnapshot();

    });
});