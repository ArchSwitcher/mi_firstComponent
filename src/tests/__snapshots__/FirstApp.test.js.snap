// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`make tests in component FirstApp should be show <FirstApp /> success 1`] = `
<Fragment>
  <h1>
    Hi im Arch Switcher
  </h1>
  <p>
     
    im a subtitle
  </p>
</Fragment>
`;
